core/src/main/java/nl/pim16aap2/bigDoors/moveBlocks/BlockMover.java

```diff
@@ -125,31 +129,33 @@ protected synchronized void putBlocks(
             byte matByte = savedBlock.getBlockByte();
             Location newPos = locationUpdater.apply(savedBlock.getRadius(), savedBlock.getStartX(),
                                                     savedBlock.getStartY(), savedBlock.getStartZ());
-
-            if (!instantOpen)
-                savedBlock.getFBlock().remove();
-
-            if (!savedBlock.getMat().equals(Material.AIR))
-            {
-                if (BigDoors.isOnFlattenedVersion())
-                {
-                    savedBlock.getBlock().putBlock(newPos);
-                    Block b = world.getBlockAt(newPos);
-                    BlockState bs = b.getState();
-                    bs.update();
+            BigDoors.getScheduler().runTask(newPos, () -> {
```

Comment:
Also wondering here why it's scheduled to run later

Reply:
Block#getState method must be called on the region thread